src/src/single-spa-angular.ts

```diff
@@ -109,23 +116,33 @@ async function mount(opts: SingleSpaAngularOpts, props: any): Promise<NgModuleRe
     null,
   );
 
+  const ngZoneEnabled = opts.NgZone !== 'noop';
+
   // The user has to provide `BrowserPlatformLocation` only if his application uses routing.
   // So if he provided `Router` but didn't provide `BrowserPlatformLocation` then we have to inform him.
-  if (opts.Router && singleSpaPlatformLocation === null) {
+  // Also `getSingleSpaExtraProviders()` function should be called only if the doesn't use
+  // `zone-less` change detection, if `NgZone` is `noop` then we can skip it.
+  if (ngZoneEnabled && opts.Router && singleSpaPlatformLocation === null) {
     throw new Error(`	
       single-spa-angular: could not retrieve extra providers from the platform injector. Did you call getSingleSpaExtraProviders() when creating platform?	
     `);
   }
 
   const bootstrappedOpts = opts as BootstrappedSingleSpaAngularOpts;
-  const ngZone: NgZone = module.injector.get(opts.NgZone);
 
-  singleSpaPlatformLocation!.setNgZone(ngZone);
-  bootstrappedOpts.bootstrappedNgZone = ngZone;
-  bootstrappedOpts.bootstrappedNgZone['_inner']._properties[bootstrappedOpts.zoneIdentifier] = true;
-  window.addEventListener('single-spa:routing-event', bootstrappedOpts.routingEventListener);
-  bootstrappedOpts.bootstrappedModule = module;
+  if (ngZoneEnabled) {
+    const ngZone: import('@angular/core').NgZone = module.injector.get(opts.NgZone);
+    // `NgZone` can be enabled but routing may not be used thus `getSingleSpaExtraProviders()`
+    // function was not called.
+    singleSpaPlatformLocation?.setNgZone(ngZone);
+    bootstrappedOpts.bootstrappedNgZone = ngZone;
+    bootstrappedOpts.bootstrappedNgZone['_inner']._properties[
+      bootstrappedOpts.zoneIdentifier
+    ] = true;
+    window.addEventListener('single-spa:routing-event', bootstrappedOpts.routingEventListener!);
```

Comment:
Have you confirmed whether routing works with zone-less applications + single-spa? The main thing to test would be if there are two separate applications active at the same time, do they both update when one of them triggers navigation.